test/base/src/androidTest/kotlin/io/realm/test/shared/RealmSchemaTests.kt

```diff
@@ -228,6 +229,47 @@ public class RealmSchemaTests {
         realm.close()
     }
 
+    @Test
+    fun dataClass() {
```

Comment:
Nice 👍 

However, I'm slightly concerned about some of the semantics of data classes and I think we need to have tests for these. Also so we can document if data classes behave differently than normal objects:

- `DataClass.copy()`: What state is the new copy... if our internal properties are copied, it means that the copy is also managed and you are actually modifying both objects. If not, this is effectively a `copyFromRealm()` method.

- What does `toString()` output? I'm not sure if we are providing a toString method currently, but since we are doing it in Java, I suspect we might end up doing it in Kotlin as well.

- What semantics does `equals()` have? In Java we have a custom implementation, I cannot remember what we do in Kotlin, but data class semantics differ from normal object equals.